src/crimson/osd/pg.cc

```diff
@@ -389,7 +389,9 @@ void PG::on_replica_activate()
 
 void PG::on_activate_complete()
 {
-  wait_for_active_blocker.unblock();
```

Comment:
Let's add a comment to explain why this is necessary:

```
/* Confusingly, on_activate_complete is invoked when the primary and replicas
 * have recorded the current interval.  At that point, the PG may either become
 * ACTIVE or PEERED, depending on whether the acting set is eligible for client
 * IO.  Only unblock wait_for_active_blocker if we actually became ACTIVE */
```

Reply:
Done:-)